### ISSUE TITLE
Can't use to Publish using Release Trigger Event

### ISSUE THREAD

**Issue by AllanOricil** [nwtgck/actions-netlify#102]
### Desired Behavior
Be able to Deploy and Publish when using with a Release Trigger Event

### What is happening
Hi I tried to use it for a Release Action Event to Deploy + Publish in Netlify. But because the isDraft is always true, It will never publish. 
```yaml
on:
  release:
    types: [published]
```

The isDraft is always true when using with a Release Action Event because of this condition
const isDraft: boolean = context.ref !== `refs/heads/${productionBranch}`

### Suggestion
I have a hotfix but I don't have permissions to push a branch and open a PR. Could you give me the permission? If not, can you add this to the conidtion

```typescript
// isDraft = true WHEN
// github.ref != production_branch
// OR
// github.ref CONTAINS tags AND github.event.release.draft == false  (This condition allow us to deploy + publish when it is on a release event)
const isDraft: boolean = context.ref !== `refs/heads/${productionBranch}` || (context.ref.includes('tags') && !context.ref.event.release.draft)
```

The contex.ref.event.release.draft I got from this doc: [Events Object](https://developer.github.com/v3/activity/events/types/#releaseevent)

And to check the context.ref for a release I just printed it with echo
![image](https://user-images.githubusercontent.com/55927613/80165447-da83d580-85b1-11ea-90a5-ca495b898fda.png)

**Comment by nwtgck** [comment]
@AllanOricil Thank you very much for your suggestion.
I'd like to consider this topic. Maybe it's is better to have another "input" rather than `production-branch: ...`. The "input" might be "production-deploy-on-release: true" (I'd like to have better and shorter name).

**Comment by jsmith** [comment]
Not the OP but I think the first idea would be great! It seems like it would be very easy to implement and would suite my needs perfectly :)

**Comment by nwtgck** [comment]
Thanks @jsmith .

## Idea 3

`$ACTIONS_NETLIFY_DEPLOY` env variable. When `$ACTIONS_NETLIFY_DEPLOY` exists, deployment is always triggered. This is also general solution for deployment.

I prefer to use general solution to avoid increasing options or inputs for flexibility and composability.

**Comment by AllanOricil** [comment]
I think the best word is "release". Im not used to have releases using other branches rather than master. If I tag and release using the Master branch, and release = true, than it is not a Draft. But go ahead and implement the best way you think and then share with us!
Thanks for implementing this action.

**Comment by nwtgck** [comment]
@AllanOricil 
Thank you very much for your comment! I also think "release" is smart, short and meaningful.

But, I'd like include "production-deploy" because Netlify officially uses "production branch" as follows.

<img width="512" alt="image" src="https://user-images.githubusercontent.com/10933561/81337971-d3e37b00-90e6-11ea-92f2-7f531bf14ed4.png">

It is easy for users to understand **"production-branch"** corresponds to  **"production-deploy"**. When I use "release", I'll expect "release-branch" not "production-branch"

**Comment by AllanOricil** [comment]
Release is not for a branch. Is when you want to release something to Production. Imagine that you deployed something to Production, but it is still not time to Release it, you have to hold. Usually we never leave merges to Production to be released automatically, we wait for a Release date. It is when all the features that has been developed are going to be published to users in production. 
What I'm saying is that it is not because I merged to the Production Branch that I want to make these changes available in Production. I only want it when I create a Release inside Github, which would trigger the "Release Event" and then start the deploy+publish to Netlify.
You have to treat these features separated they mean different things.

**Comment by jsmith** [comment]
@nwtgck will have more to say but I think this `Production branch` concept is really a just a netlify thing. I don't think it's a concept that is really useful for production environments but it is what it is! I do think the `production-branch` option should be better documented though since it's not very clear to users of this action what this option actually does.

**Comment by nwtgck** [comment]
@AllanOricil 
> Release is not for a branch. Is when you want to release something to Production.

Yes. I think so too. My previous  comment is about name of option/flag to solve this issue. So I was thinking about what better name is. How about the following input?

```yaml
with:
  - production-deploy: ${{ some condition }} 
```

**Comment by nwtgck** [comment]
@jsmith Thanks. `production-branch` is just a branch name which can be released as production when pushed to the branch.

**Comment by AllanOricil** [comment]
I think it is better to not use words related to an specific environment. Think that we want to use this action for any of our Netlify Environments. I can have 3 Netlify environments in my dev lifecycle (DEV, QA, Prod) and in each I want to be able to use your action to Deploy, Deploy and Publish or just Publish (when it is a Release Event).

**Comment by nwtgck** [comment]
@AllanOricil Thanks. Do you mean name `production-deploy` is related to an specific environment? My expectation of the input is like the following.

```yaml
with:
  - production-deploy: ${{ !(context.ref.includes('tags') && !context.ref.event.release.draft) }}
```

**Comment by AllanOricil** [comment]
yes, use properties that are environment agnostic. 
I can use it for any of my environments. And it is clear for those who use Netlify that it requires a branch to get the data from. Is there any other branch parameter in Netlify? If not just leave as Branch and people will figure out. 

````yaml
with:
          publish-dir: './dist'
          branch: master //IT IS THE BRANCH I WANT NETLIFY TO GET THE DIST FOLDER FROM
          github-token: ${{ secrets.GITHUB_TOKEN }}
          deploy-message: "Deploy from GitHub Actions"
          enable-pull-request-comment: true
          enable-commit-comment: true
          overwrites-pull-request-comment: true
````

**Comment by nwtgck** [comment]
@AllanOricil 

>  If not just leave as Branch and people will figure out.

I don't think so. I think name `production-branch` is clear for users.
Any branch can trigger deploys to Netlify, but the deploy is not production. For example,  when you have `my-feature1` and `my-feature` branches, they will trigger deploy to "my-feature1--hogehoge.netlify.app", "my-feature2--hogehoge.netlify.app" However, only pushing to `production-branch` triggers a production deploy to "hogehoge.netlify.app".

So, I decided to name `production-branch` as follows.
<img width="489" alt="image" src="https://user-images.githubusercontent.com/10933561/81820732-f1439980-956b-11ea-9976-08f38b684cc6.png">

But, in this issue, we should discuss  what input name is better for production deploy.

**Comment by AllanOricil** [comment]
What if I decide to use your action to deploy to my QA environment? Do you see sense using the name production-branch?
Actions should not have environment specific names because we can use them in multiple environments.

**Comment by nwtgck** [comment]
@AllanOricil 

> What if I decide to use your action to deploy to my QA environment? Do you see sense using the name production-branch?

Yes, I understand what you mean. "Production" in QA might confuse users. 

However, this action is about Netlify. "Production branch" is the term of Netlify. When you go to Netlify settings > Deploy settings, you will find "Production branch" as follows.

<img width="569" alt="image" src="https://user-images.githubusercontent.com/10933561/82101626-ee9d9b80-9747-11ea-9a92-3b7e5379ee19.png">

The terms should be used the same term as Netlify.

**Comment by AllanOricil** [comment]
I see. They probably made a mistake in this page were you took the ps. When you launch a brand new project it says only branch.

![image](https://user-images.githubusercontent.com/55927613/82105393-09a1e880-9713-11ea-91e1-5d02a388baf8.png)

**Comment by nwtgck** [comment]
OK. I understand Netlify uses two terms for the same meaning.

**Comment by AllanOricil** [comment]
It is really weird. They call it Production but it is the only branch you can configure. Why would you put  a name on it if it is the only option. Maybe there is a paid version where you can configure many different branches. Do you know anything about it? I only used the free version.

**Comment by nwtgck** [comment]
> Do you know anything about it? I only used the free version.

I don't know and I had paid before but I don't.

---

> They call it Production but it is the only branch you can configure. Why would you put a name on it if it is the only option. 

Maybe "only option", but I may understand Netlify and what Netlify means. Netlify deploys every branch, not only "master". That is a strong advantage of Netlify comparing with other static deployment service such as GitHub Pages. **Ever branch is deployed** with a prefix. For example, piping-ui.netlify.app is "Production branch". Netlify deploys other branches with prefixes as follows.
- `develop` branch:
  - https://develop--piping-ui.netlify.app/
- `feature/passworless-protection` branch:
  - https://feature-passwordless-protection--piping-ui.netlify.app/
- `feature/target-es2015` branch:
  - https://feature-target-es2015--piping-ui.netlify.app/

**Comment by AllanOricil** [comment]
I will be waiting for the release! Thanks @nwtgck

**Comment by nwtgck** [comment]
@AllanOricil Thanks!

But, I didn't think the naming problem is not solved. Maybe I'll use "production deploy". However, I understand well that "production" is a confusing word, thanks to you. At some point, we can use other proper words and deprecate use of "production-ooo" inputs in major changes.

**Comment by AllanOricil** [comment]
I will reopen this. 
Once you implement, comment here so we can close.

**Comment by nwtgck** [comment]
>  Once you implement, comment here so we can close.

I also think it's better.

**Comment by kwkbtr** [comment]
:1 for `production-deploy` option in https://github.com/nwtgck/actions-netlify/issues/102#issuecomment-627651839 👍

**Comment by gvdp** [comment]
Hi, I was running into the same issue so I tried to implement this option in https://github.com/nwtgck/actions-netlify/pull/188

**Comment by nwtgck** [comment]
The feature is available now with `@v1`, `@v1.1`, `@v1.1.3`!
(docs: https://github.com/nwtgck/actions-netlify/tree/6a4dc6d96b50778c6aca63ee59bcc6f5aa07b745#optional-inputs)
